src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -168,23 +168,27 @@ export const getBookInformation = async(
 
   const configuredReference = allReferences.filter((item) => BOOKS[item.id])[0];
 
-  if (configuredReference) {
-    const osWebBook =  await services.osWebLoader.getBookFromId(configuredReference.id);
-    const archiveBook = await services.archiveLoader.book(
-      configuredReference.id, BOOKS[configuredReference.id].defaultVersion
-    ).load();
-
-    return {osWebBook, archiveBook};
-
-  } else if (UNLIMITED_CONTENT) {
-    for (const {id, bookVersion} of allReferences) {
-      const osWebBook =  await services.osWebLoader.getBookFromId(id).catch(() => undefined);
-      const archiveBook = await services.archiveLoader.book(id, bookVersion).load();
-      if (archiveBook && archiveTreeSectionIsBook(archiveBook.tree)) {
-        return {osWebBook, archiveBook};
+  try {
```

Comment:
this is a very aggressive try/catch, i think it would also work to change the one line (182 in this diff)

```
const archiveBook = await services.archiveLoader.book(id, bookVersion).load();
```

to

```
const archiveBook = await services.archiveLoader.book(id, bookVersion).load().catch(() => undefined);
```

this way the change will only affect the one query we actually expect to fail within the `UNLIMITED_CONTENT` portion of the logic

Reply:
For the page that you've posted in the slack: `/books/university-physics-volume-2@86cddf1/pages/1-6-mechanisms-of-heat-transfer?archive=https%3A%2F%2Fopenstax.org%2Fapps%2Farchive-preview%2F20210208.162118` it is failing for lines 174-176 - so it is in `configuredReference`

Will it be okay to remove `try catch` and replace the whole block with: ?
```ts
if (configuredReference) {
    const osWebBook =  await services.osWebLoader.getBookFromId(configuredReference.id);
    const archiveBook = await services.archiveLoader.book(
      configuredReference.id, BOOKS[configuredReference.id].defaultVersion
    ).load().catch(() => undefined);

    if (archiveBook && archiveTreeSectionIsBook(archiveBook.tree)) {
      return {osWebBook, archiveBook};
    }
  } else if (UNLIMITED_CONTENT) {
    for (const {id, bookVersion} of allReferences) {
      const osWebBook =  await services.osWebLoader.getBookFromId(id).catch(() => undefined);
      const archiveBook = await services.archiveLoader.book(id, bookVersion).load().catch(() => undefined);
      if (archiveBook && archiveTreeSectionIsBook(archiveBook.tree)) {
        return {osWebBook, archiveBook};
      }
    }
  }
```

So we add `.catch(() => undefined)` in both branches of if statement and in the first one we also return only if `archiveBook && archiveTreeSectionIsBook(archiveBook.tree)`

Reply:
uhhh let me think this through...

- the `getBookInformation` is only used for other-book links
- web-pipeline content doesn't have a version in the link, the version comes from the book config in `getInputReferenceInfo`
- so the `else if (UNLIMITED_CONTENT) {` block would never be hit for web-pipeline content, it would only be used for cnx.org content that was loading arbitrary book ids through `archiveLoader.getBookIdsForPage`

so lets:
- do the change with the catch on `archiveLoader` within `UNLIMITED_CONTENT` as we discussed already
- also add a comment at the top of that block `//this section only used for cnx.org content, delete when web-pipeline (RAP) is adopted

and:
- add the catch to `services.archiveLoader` in the `configuredReference` block as you described, but instead of just returning undefined, rethrow the error if `UNLIMITED_CONTENT` is unset. (because this logic is also used in production)